src/extensions/views/EditPluginExtension/EditPluginExtension.test.ts

```diff
@@ -0,0 +1,83 @@
+// @ts-strict-ignore
+import {
+  ConfigurationItemFragment,
+  ConfigurationItemInput,
+  ConfigurationTypeFieldEnum,
+} from "@dashboard/graphql";
+
+import { getConfigurationInput } from "./EditPluginExtension";
+
+const baseConfig: Omit<ConfigurationItemFragment, "name" | "type" | "value"> = {
+  __typename: "ConfigurationItem",
+  helpText: "",
+  label: "",
+};
+const config: ConfigurationItemFragment[] = [
+  {
+    ...baseConfig,
+    name: "field-1",
+    type: ConfigurationTypeFieldEnum.STRING,
+    value: "val1",
+  },
+  {
+    ...baseConfig,
+    name: "field-2",
+    type: ConfigurationTypeFieldEnum.STRING,
+    value: "val2",
+  },
+  {
+    ...baseConfig,
+    name: "field-3",
+    type: ConfigurationTypeFieldEnum.PASSWORD,
+    value: "",
+  },
+  {
+    ...baseConfig,
+    name: "field-4",
+    type: ConfigurationTypeFieldEnum.SECRET,
+    value: "val4",
+  },
+];
+const input: ConfigurationItemInput[] = [
+  {
+    name: "field-1",
+    value: "value1",
+  },
+  {
+    name: "field-2",
+    value: "value2",
+  },
+  {
+    name: "field-3",
+    value: "value3",
+  },
+  {
+    name: "field-4",
+    value: "value4",
+  },
+];
+
+test("Ensure that no secret is sent in input", () => {
+  const output: ConfigurationItemInput[] = getConfigurationInput(config, input);
+
+  expect(output).toHaveLength(2);
+  expect(
+    output.find(
+      field =>
+        config.find(configField => configField.name === field.name).type ===
+        ConfigurationTypeFieldEnum.PASSWORD,
```

Comment:
the test cases should not contain any specific logic, i think you should make just toEqual call